Classes/EventListener/ProcessRequestTokenListener.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+namespace Causal\Oidc\EventListener;
+
+use TYPO3\CMS\Core\Authentication\Event\BeforeRequestTokenProcessedEvent;
+use TYPO3\CMS\Core\Security\RequestToken;
+
+final class ProcessRequestTokenListener
+{
+    public function __invoke(BeforeRequestTokenProcessedEvent $event): void
+    {
+        $user = $event->getUser();
+        $requestToken = $event->getRequestToken();
+        if ($requestToken instanceof RequestToken) {
+            // fine, there is a valid request-token
+            return;
+        }
+        // @TODO how to improve security? Maybe get rid of redirect in \Causal\Oidc\Controller\AuthenticationController::connectAction?
```

Comment:
This controller does not even exist anymore.
Not sure why we need this listener here at all?

Reply:
the feature is introduced here and ist about CSRF:
[Feature-97305-IntroduceCSRF-likeRequest-tokenHandling](https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Feature-97305-IntroduceCSRF-likeRequest-tokenHandling.html#:~:text=Description,a%20%22pre%2Dsession%22.)

And its called in AbstractUserAuthentication::checkAuthentication and stops a valid login in some cases if the token is noch present.
But i have just rebased the code and i am not that familiar with all the login requirements right now

Reply:
according to the controller, yes the controller i even there, but the query param "tx_oidc" is still added to redirect urls and the redirect is done by the middleware "AuthenticationUrlRequest" or the "LoginController"

i will remove the hint to AuthenticationController

Reply:
Thanks for the link. I wasn't aware this is called in AbstractUserAuthentication too. Thanks for the update too.